Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -179,409 +135,356 @@ public void Dispose()
+                    var ips = remoteIPFilter.Where(x => !x.Contains('/', StringComparison.OrdinalIgnoreCase));
+                    foreach (var ip in ips)
                     {
-                        if (intf.Contains(source))
+                        if (IPAddress.TryParse(ip, out var ipp))
                         {
-                            result = FormatIP6String(intf.Address);
-                            _logger.LogDebug("{Source}: GetBindInterface: Has source, matched best internal interface on range. {Result}", source, result);
-                            return result;
+                            _remoteAddressFilter.Add(new IPNetwork(ipp, ipp.AddressFamily == AddressFamily.InterNetwork ? 32 : 128));
                         }
                     }
                 }
-
-                result = FormatIP6String(interfaces.First().Address);
-                _logger.LogDebug("{Source}: GetBindInterface: Matched first internal interface. {Result}", source, result);
-                return result;
             }
-
-            // There isn't any others, so we'll use the loopback.
-            result = IsIP6Enabled ? "::1" : "127.0.0.1";
-            _logger.LogWarning("{Source}: GetBindInterface: Loopback {Result} returned.", source, result);
-            return result;
         }
 
-        /// <inheritdoc/>
-        public Collection<IPObject> GetInternalBindAddresses()
+        /// <summary>
+        /// Parses the user defined overrides into the dictionary object.
+        /// Overrides are the equivalent of localised publishedServerUrl, enabling
+        /// different addresses to be advertised over different subnets.
+        /// format is subnet=ipaddress|host|uri
+        /// when subnet = 0.0.0.0, any external address matches.
+        /// </summary>
+        private void InitialiseOverrides(NetworkConfiguration config)
         {
-            int count = _bindAddresses.Count;
-
-            if (count == 0)
+            lock (_initLock)
             {
-                if (_bindExclusions.Count > 0)
-                {
-                    // Return all the internal interfaces except the ones excluded.
-                    return CreateCollection(_internalInterfaces.Where(p => !_bindExclusions.ContainsAddress(p)));
-                }
-
-                // No bind address, so return all internal interfaces.
-                return CreateCollection(_internalInterfaces);
-            }
-
-            return new Collection<IPObject>(_bindAddresses.Where(a => IsInLocalNetwork(a)).ToArray());
-        }
-
-        /// <inheritdoc/>
-        public bool IsInLocalNetwork(IPObject address)
-        {
-            return IsInLocalNetwork(address.Address);
-        }
-
-        /// <inheritdoc/>
-        public bool IsInLocalNetwork(string address)
-        {
-            return IPHost.TryParse(address, out IPHost ipHost) && IsInLocalNetwork(ipHost);
-        }
+                _publishedServerUrls.Clear();
```

Comment:
Same as above. The over all state of the network manager is not valid when we alter fields that aren't synchronized

Reply:
How could that be properly addressed?